compiler-lib/src/optimization/constant_folding.rs

```diff
@@ -257,25 +369,39 @@ impl ConstantFolding {
                     },
                 )
             }
+
+            // == Value nodes ==
             _ => {
```

Comment:
No need to change this. But you could reduce the nesting here by matching on `_ if condition =>`.

Reply:
Ah I see. I feared that rust wouldn't do a jump table anymore with guarded matches.